apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/McsClaimTransformerTest.java

```diff
@@ -0,0 +1,484 @@
+package gov.cms.bfd.pipeline.rda.grpc.source;
+
+import static gov.cms.bfd.pipeline.rda.grpc.source.TransformerTestUtils.assertListContentsHaveSamePropertyValues;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.samePropertyValuesAs;
+import static org.junit.Assert.*;
+
+import com.google.common.collect.ImmutableList;
+import gov.cms.bfd.model.rda.PreAdjMcsClaim;
+import gov.cms.bfd.model.rda.PreAdjMcsDetail;
+import gov.cms.bfd.model.rda.PreAdjMcsDiagnosisCode;
+import gov.cms.bfd.pipeline.rda.grpc.RdaChange;
+import gov.cms.bfd.pipeline.sharedutils.IdHasher;
+import gov.cms.mpsm.rda.v1.ClaimChange;
+import gov.cms.mpsm.rda.v1.mcs.McsBeneficiarySex;
+import gov.cms.mpsm.rda.v1.mcs.McsBillingProviderStatusCode;
+import gov.cms.mpsm.rda.v1.mcs.McsClaim;
+import gov.cms.mpsm.rda.v1.mcs.McsClaimType;
+import gov.cms.mpsm.rda.v1.mcs.McsDetail;
+import gov.cms.mpsm.rda.v1.mcs.McsDetailStatus;
+import gov.cms.mpsm.rda.v1.mcs.McsDiagnosisCode;
+import gov.cms.mpsm.rda.v1.mcs.McsDiagnosisIcdType;
+import gov.cms.mpsm.rda.v1.mcs.McsStatusCode;
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import org.junit.Before;
+import org.junit.Test;
+
+public class McsClaimTransformerTest {
+  // using a fixed Clock ensures our timestamp is predictable
+  private final Clock clock = Clock.fixed(Instant.ofEpochMilli(1621609413832L), ZoneOffset.UTC);
+  private final McsClaimTransformer transformer =
+      new McsClaimTransformer(
+          clock,
+          new IdHasher(
+              new IdHasher.Config(1000, "nottherealpepper".getBytes(StandardCharsets.UTF_8))));
+  private ClaimChange.Builder changeBuilder;
+  private McsClaim.Builder claimBuilder;
+  private PreAdjMcsClaim claim;
+
+  @Before
+  public void setUp() {
+    changeBuilder = ClaimChange.newBuilder();
+    claimBuilder = McsClaim.newBuilder();
+    claim = new PreAdjMcsClaim();
+  }
+
+  @Test
+  public void minimumValidClaim() {
+    claim.setIdrClmHdIcn("123456789012345");
+    claim.setIdrContrId("12345");
+    claim.setIdrClaimType("3");
+    claim.setLastUpdated(clock.instant());
+    claimBuilder
+        .setIdrClmHdIcn("123456789012345")
+        .setIdrContrId("12345")
+        .setIdrClaimTypeEnum(McsClaimType.CLAIM_TYPE_MEDICAL);
+    changeBuilder
+        .setChangeType(ClaimChange.ChangeType.CHANGE_TYPE_INSERT)
+        .setMcsClaim(claimBuilder.build());
+    assertChangeMatches(RdaChange.Type.INSERT);
```

Comment:
Are these testing the entire transformer class?

Reply:
This is testing that a claim containing only the required fields can be transformed without errors.  So it verifies that no extra fields are required by the transformer for some reason.